service/history/api/recordchildworkflowcompleted/api.go

```diff
@@ -153,6 +157,28 @@ func recordChildWorkflowCompleted(
 				return nil, consts.ErrChildExecutionNotFound
 			}
 
+			childrenInitializedAfterResetPoint := mutableState.GetChildrenInitializedPostResetPoint()
+			if len(childrenInitializedAfterResetPoint) > 0 {
+				// This parent was reset and it also has some children that potentially were restarted.
+				initiatedEvent, err := mutableState.GetChildExecutionInitiatedEvent(ctx, parentInitiatedID)
+				if err != nil {
+					return nil, consts.ErrChildExecutionNotFound
+				}
+				initiatedAttr := initiatedEvent.GetStartChildWorkflowExecutionInitiatedEventAttributes()
+				childID := fmt.Sprintf("%s:%s", initiatedAttr.GetWorkflowType().Name, initiatedAttr.GetWorkflowId())
```

Comment:
nit: we really should create a util function for this concatenation logic. I think I've seen it in 3 different places.